crates/api/src/local_user/save_settings.rs

```diff
@@ -117,6 +119,21 @@ impl Perform for SaveUserSettings {
     })
     .await?
     .map_err(|e| LemmyError::from_error_message(e, "user_already_exists"))?;
+    let discussion_languages: Option<Vec<LanguageId>> =
+      if let Some(discussion_languages) = data.discussion_languages.clone() {
+        let mut language_ids = vec![];
+        for l in discussion_languages {
+          language_ids.push(
+            blocking(context.pool(), move |conn| {
+              Language::read_id_from_code(conn, l)
+            })
+            .await??,
+          );
+        }
+        Some(language_ids)
+      } else {
+        None
+      };
```

Comment:
This should be inserting / upserting into the `local_user_language` table, which is still missing.

Reply:
Also, as weird as it sounds, the API should probably take in a vector of `language_id`: 

- Its super easy to get those, since they will come back with GetSite.
- The `local_user_language` only needs (local_user_id, language_id), so you can do upserts there without have to do a pointless select from the language table.